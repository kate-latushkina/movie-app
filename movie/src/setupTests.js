// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Enzyme, { shallow, render, mount } from "enzyme";
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

global.shallow = shallow;
global.render = render;
global.mount = mount;
global.toJson = toJson;

// Fail tests on any warning
console.error = message => {
  throw new Error(message);
};
