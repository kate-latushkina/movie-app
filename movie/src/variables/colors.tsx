const colors = {
  darkViolet: "#593158",
  darkGrey: "#424242",
  lightGrey: "#555555",
  pink: "#f65261",
  white: "#fff",
};

export default colors;
