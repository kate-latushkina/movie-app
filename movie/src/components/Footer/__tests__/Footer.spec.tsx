import React from "react";
import renderer from "react-test-renderer";
import FooterSection from "../Footer";
import "../../../setupTests";
import colors from "../../../variables/colors";
import "jest-styled-components";

describe("should render Footer component", () => {
  it("should contain FooterSection", () => {
    const wrapper = renderer.create(<FooterSection />).toJSON();
    expect(wrapper).toHaveStyleRule("background-color", colors.darkGrey);
  });
});
