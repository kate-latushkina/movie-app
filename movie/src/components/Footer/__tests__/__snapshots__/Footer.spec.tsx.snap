// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Footer component should render FooterSection snapshot 1`] = `
.c1 {
  color: #f65261;
}

.c0 {
  background-color: #424242;
  padding: 20px 0;
}

<div
  className="MuiGrid-root c0 MuiGrid-container MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root c1"
  >
    <b>
      netflix
    </b>
    <span>
      roulette
    </span>
  </div>
</div>
`;
